src/pull.go

```diff
@@ -82,6 +88,72 @@ func (g *Commands) Pull(byId bool) error {
 	return g.playPullChanges(nonConflicts, g.opts.Exports, opMap)
 }
 
+func autoRenameClashes(g *Commands, clashes []*Change) error {
+	type rename struct {
+		name string
+		change *Change
+	}
+	clashesMap := map[string][]rename{}
+
+	for _, clash := range clashes {
+		group := clashesMap[clash.Path]
+		if clash.Src == nil { // Is that even possible? In what situation?
+			continue
+		}
+		clashesMap[clash.Path] = append(group, rename{"", clash})
+	}
+
+	for path, group := range clashesMap {
+		ext := filepath.Ext(path)
+		name := strings.TrimSuffix(path, ext)
+		n := 0
+
+		for i, r := range group[1:] {  // we can leave first item with original name
+			var newName string
+			for {
+				newName = fmt.Sprintf("%v_%d%v", name, n, ext)
+				n++
```

Comment:
Oh okay, I see. We could adapt it though :) Sorry for that. I just want to avoid too much book keeping. Also tomorrow evening I'll be able to actually run and test the changes on a computer. This is because mostly when in class I can only use a tablet or browser or another computer without a compiler. Since tomorrow I'll be done early, I can jump in.